core/src/test/java/org/bitcoinj/crypto/ChildKeyDerivationTest.java

```diff
@@ -171,9 +171,15 @@ public void encryptedDerivation() throws Exception {
     @Test
     public void pubOnlyDerivation() throws Exception {
         DeterministicKey key1 = HDKeyDerivation.createMasterPrivateKey("satoshi lives!".getBytes());
+        assertFalse(key1.isPubKeyOnly());
         DeterministicKey key2 = HDKeyDerivation.deriveChildKey(key1, ChildNumber.ZERO_HARDENED);
+        assertFalse(key2.isPubKeyOnly());
         DeterministicKey key3 = HDKeyDerivation.deriveChildKey(key2, ChildNumber.ZERO);
-        DeterministicKey pubkey3 = HDKeyDerivation.deriveChildKey(key2.getPubOnly(), ChildNumber.ZERO);
+        assertFalse(key3.isPubKeyOnly());
+        DeterministicKey pubkey2 = key2.getPubOnly();
+        assertFalse(pubkey2.isPubKeyOnly());
```

Comment:
But the line you're quoting exhibits the problem I'm stuck with. Yes, the key should be public only. But it isn't because it contains the reference to a private parent. Someone has to make decisions here – I can't do that.